src/workspace/discovery.rs

```diff
@@ -607,33 +464,290 @@ fn discover_workspace_config_files_for_single_dir(
+              include: Some(PathOrPatternSet::new(patterns)),
+              exclude: PathOrPatternSet::new(Vec::new()),
+            },
+          )
+          .map_err(|err| {
+            WorkspaceDiscoverErrorKind::FailedCollectingNpmMembers {
+              package_json_url: pkg_json.specifier(),
+              source: err,
+            }
+          })?
+      };
+      let mut member_dir_urls =
+        IndexSet::with_capacity(path_members.len() + pkg_json_paths.len());
+      for path_member in path_members {
+        let member_dir_url = resolve_member_url(path_member)?;
+        member_dir_urls.insert(member_dir_url);
+      }
+      for pkg_json_path in pkg_json_paths {
+        let member_dir_url =
+          Url::from_directory_path(pkg_json_path.parent().unwrap()).unwrap();
+        member_dir_urls.insert(member_dir_url);
+      }
+
+      for member_dir_url in member_dir_urls {
+        if member_dir_url == root_config_file_directory_url {
+          continue; // ignore self references
+        }
+        validate_member_url_is_descendant(&member_dir_url)?;
+        let member_config_folder =
+          match find_member_config_folder(&member_dir_url) {
+            Ok(config_folder) => config_folder,
+            Err(ResolveWorkspaceMemberError::NotFound { dir_url }) => {
+              // enhance the error to say we didn't find a package.json
+              return Err(
+                ResolveWorkspaceMemberError::NotFoundPackageJson { dir_url }
+                  .into(),
+              );
+            }
+            Err(err) => return Err(err.into()),
+          };
+        if member_config_folder.pkg_json().is_none() {
+          return Err(
+            ResolveWorkspaceMemberError::NotFoundPackageJson {
+              dir_url: member_dir_url,
+            }
+            .into(),
+          );
+        }
+        // don't surface errors about duplicate members for
+        // package.json workspace members
+        final_members.insert(new_rc(member_dir_url), member_config_folder);
+      }
+    }
+  }
+
+  let workspace = new_rc(Workspace::new_with_members(
+    root_config_folder,
+    final_members,
+  ));
+
+  if let Some(cache) = opts.workspace_cache {
+    cache.set(workspace.root_dir_path(), workspace.clone());
+  }
+
+  Ok(workspace)
+}
+
+fn resolve_vendor_dir(
+  maybe_deno_json: Option<&ConfigFile>,
+  maybe_vendor_override: Option<&VendorEnablement>,
+) -> Option<PathBuf> {
+  if let Some(vendor_folder_override) = maybe_vendor_override {
+    match vendor_folder_override {
+      VendorEnablement::Disable => None,
+      VendorEnablement::Enable { cwd } => match maybe_deno_json {
+        Some(c) => Some(c.dir_path().join("vendor")),
+        None => Some(cwd.join("vendor")),
+      },
+    }
+  } else {
+    let deno_json = maybe_deno_json?;
+    if deno_json.vendor() == Some(true) {
+      Some(deno_json.dir_path().join("vendor"))
+    } else {
+      None
+    }
   }
 }
```

Comment:
Changed this to just accept a deno json instead of ConfigFolders which was needlessly complex